### ISSUE TITLE
Pretty-printing doesn't handle Word

### ISSUE THREAD

**Issue by neongreen** [well-typed/binary-serialise-cbor#106]
``` haskell
> show $ encode (5 :: Word)
"[TkInt 5]"

> prettyHexEnc $ encode (5 :: Word)
"pprint: Unexpected token: TkWord 5 TkEnd"
```

(Why does it say `TkWord` when the `Show` instance gives `TkInt`?)

**Comment by thoughtpolice** [comment]
Thanks -- the pretty printer indeed doesn't handle the `TkWord` case properly it seems.

The reason for the output discrepancy is because the `Show` instance for `Encoding` actually uses the `FlatTerm` module, who's AST type does not distinguish between `Word` encodings and `Int` encodings (it's a simpler representation meant for validity testing). The `Encoding` type itself *does* distinguish these cases, which is why you see this.

In short, you're looking at two different constructors. I agree this behavior is confusing, though, and I'm hoping to fix some of it once #104 is fixed.

**Comment by axman6** [comment]
Oops, looks like this was my screwup (which would've been called if I weren't lazy and used `_ -> fail`. If I get time in the next few days I'll fix it, but it is trivial enough to fix that I would be very happy if someone else does it!

**Comment by bgamari** [comment]
Fixed.

**Comment by axman6** [comment]
Thanks Ben!
